lib/site_prism/element_container.rb

```diff
@@ -228,14 +228,36 @@ def deduce_iframe_element_find_args(args)
     end
 
     def extract_section_options(args, &block)
-      if args.first.is_a?(Class)
-        section_class = args.shift
-      elsif block_given?
-        section_class = Class.new(SitePrism::Section, &block)
-      else
-        raise ArgumentError, 'You should provide section class either as a block, or as the second argument.'
+      section_class = deduce_section_class(args, &block)
+      arguments = deduce_search_arguments(section_class, args)
+      [section_class, arguments]
+    end
+
+    def deduce_section_class(args, &block)
+      if args.first.is_a?(Class) && args.first.ancestors.include?(SitePrism::Section)
+        klass = base_class = args.shift
       end
-      [section_class, args]
+
+      if block_given?
+        klass = Class.new(base_class || SitePrism::Section, &block)
+      end
+
+      unless klass
+        raise ArgumentError, "You should provide section class either \
+as a block, or as the second argument."
+      end
+      klass
+    end
+
+    def deduce_search_arguments(section_class, args)
+      extract_search_arguments(args) ||
+        extract_search_arguments(section_class.default_search_arguments) ||
+        raise(ArgumentError, "You should provide search arguments \
+in section creation or set_default_search_arguments within section class")
+    end
+
+    def extract_search_arguments(args)
```

Comment:
I guess, this method can be used in the future to do more work, i.e. converting site_prism's search arguments to capybara's (Now they are the same match, but who knows what they'll change in the future). IMO, it's not good to have dependencies on internal implementation (capybara dsl).